internal/devices/create.go

```diff
@@ -64,62 +66,123 @@ func (c *Client) Create() *cobra.Command {
 				if readErr != nil {
-					return fmt.Errorf("Could not read userdata-file: %w", readErr)
+					return fmt.Errorf("could not read userdata-file: %w", readErr)
 				}
 				userdata = string(userdataRaw)
 			}
+			var endDt time.Time
 
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
-					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
+					return fmt.Errorf("could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
+				endDt = parsedTime
 			}
 
+			formatedCdata := make(map[string]interface{})
+			if customdata != "" {
+				err := json.Unmarshal([]byte(customdata), &formatedCdata)
+				if err != nil {
+					return fmt.Errorf("could not convert customedata :%w", err)
+				}
+			}
 			var facilityArgs []string
+
+			var request metal.ApiCreateDeviceRequest
 			if facility != "" {
 				facilityArgs = append(facilityArgs, facility)
-			}
 
-			request := &packngo.DeviceCreateRequest{
-				Hostname:              hostname,
-				Plan:                  plan,
-				Facility:              facilityArgs,
-				Metro:                 metro,
-				OS:                    operatingSystem,
-				BillingCycle:          billingCycle,
-				ProjectID:             projectID,
-				UserData:              userdata,
-				CustomData:            customdata,
-				IPXEScriptURL:         ipxescripturl,
-				Tags:                  tags,
-				PublicIPv4SubnetSize:  publicIPv4SubnetSize,
-				AlwaysPXE:             alwaysPXE,
-				HardwareReservationID: hardwareReservationID,
-				SpotInstance:          spotInstance,
-				SpotPriceMax:          spotPriceMax,
-				TerminationTime:       endDt,
+				facilityDeviceRequest := metal.CreateDeviceRequest{
+					DeviceCreateInFacilityInput: &metal.DeviceCreateInFacilityInput{
+						Facility:        facilityArgs,
+						Plan:            plan,
+						OperatingSystem: operatingSystem,
+						Hostname:        &hostname,
+						Userdata:        &userdata,
+						Tags:            tags,
+					},
+				}
+
+				if billingCycle != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetBillingCycle(billingCycle)
+				}
+				if alwaysPXE {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetAlwaysPxe(alwaysPXE)
+				}
+
+				if ipxescripturl != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetIpxeScriptUrl(ipxescripturl)
+				}
+				if publicIPv4SubnetSize != 0 {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetPublicIpv4SubnetSize(int32(publicIPv4SubnetSize))
+				}
+				if terminationTime != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetTerminationTime(endDt)
+				}
+				facilityDeviceRequest.DeviceCreateInFacilityInput.SetSpotInstance(spotInstance)
+				facilityDeviceRequest.DeviceCreateInFacilityInput.SetSpotPriceMax(float32(spotPriceMax))
+				facilityDeviceRequest.DeviceCreateInFacilityInput.SetCustomdata(formatedCdata)
```

Comment:
After some local testing (removing some attributes from the code and re-running the tests) it looks like provisioning fails when we send an empty JSON object in `customdata`.  This should be updated to only set custom data if the user provided it.